### PR TITLE
fix/rpg switch speed

### PR DESCRIPTION
**Pull request by bonjorno7**
Changed all weapons to allow you to fire instantly without having to wait for the weapon draw animation to finish.
Changed the deploy time for the rpg and shotgun to 0.5 seconds to match TF2.
Changed the shotgun fire time to 0.625 seconds to match TF2.
Changed rpg draw animation to 68 fps, so it lasts 0.5 seconds, matching the hardcoded value.
I was going to include new animations for the shotgun but that's out of scope.

### CODE REVIEW

**Comment by bonjorno7** [comment]
right so we don't add those things for pull requests, yeet

File: mp/src/game/shared/basecombatweapon_shared.h [hunk 1]
```diff
@@ -247,6 +247,7 @@ class CBaseCombatWeapon : public BASECOMBATWEAPON_DERIVED_FROM
 	
 	virtual bool			CanHolster( void ) const { return TRUE; };		// returns true if the weapon can be holstered
 	virtual bool			DefaultDeploy( char *szViewModel, char *szWeaponModel, int iActivity, char *szAnimExt );
+    virtual float           DeployTime() const { return 0.0f; }
```
Comment (Gocnak) [inline mp/src/game/shared/basecombatweapon_shared.h]: Perhaps 0 deploy time may not be best for every weapon now that I think about it. Just make 0.5 the default so people can't spam the paintgun or other weapons in lobbies.

Leave it as a virtual function though so anything can override it.

File: mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp [hunk 2]
```diff
@@ -83,13 +83,13 @@ void CMomentumShotgun::PrimaryAttack()
 #endif
         m_flPumpTime = gpGlobals->curtime + 0.5;
```
Comment (Gocnak) [inline mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp]: It's good you pulled the 0.25 out into a getter, you may want to do the same for pump time.

File: mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp [hunk 3]
```diff
@@ -83,13 +83,13 @@ void CMomentumShotgun::PrimaryAttack()
...
+    m_flNextPrimaryAttack = gpGlobals->curtime + PrimaryFireTime();
+    m_flNextSecondaryAttack = gpGlobals->curtime + PrimaryFireTime();
 #ifdef WEAPONS_USE_AMMO
     if (m_iClip1 != 0)
         SetWeaponIdleTime(gpGlobals->curtime + 2.5);
     else
-        SetWeaponIdleTime(gpGlobals->curtime + 0.25);
+        SetWeaponIdleTime(gpGlobals->curtime + PrimaryFireTime());
 #else
     SetWeaponIdleTime(gpGlobals->curtime + 2.5f);
```
Comment (Gocnak) [inline mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp]: You may want to make this 2.5f a const for the function, or a `define` up top if the value (and its context) is used outside of this function, I'm seeing it in multiple places.

File: mp/src/game/shared/basecombatweapon_shared.h (hunk 1 above)
Comment (bonjorno7) [reply mp/src/game/shared/basecombatweapon_shared.h]: Consider it done

File: mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp (hunk 2 above)
Comment (bonjorno7) [reply mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp]: Consider it done

File: mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp (hunk 3 above)
Comment (bonjorno7) [reply mp/src/game/shared/momentum/weapon/weapon_mom_shotgun.cpp]: Consider it done

**Comment by bonjorno7** [comment]
forgettaboutit

**Pull request merged**